### DISCUSSION TITLE
Presets is not optional, how do I cover plugin in presets?

### DISCUSSION THREAD

**Discussion by 7mouse** [babel/babel#12885] [General] [answered]
If I use a presets contain a plugin A that transform let to const, but I expect all code use let, I can't cover A by use plugin B that transform const to let.

```json
{
    "presets": ["presetA"],
    "plugins": ["pluginB"]
}
```

- Plugins run before Presets.
- Plugin ordering is first to last.
- Preset ordering is reversed (last to first).

How do I turn off some of its plugins while using the presets? Thanks!

**Comment by nicolo-ribaudo** [comment, marked as answer]
You can only do of the preset provides an option to do so, otherwise you need to fork and modify the preset.

**Reply by 7mouse** [reply]
Can I use like this? Thank you!
```json
{
  "presets": [
    {
      "plugins": ["pluginB"]
    }, "presetA"
  ]
}
```

**Reply by 7mouse** [reply]
because we can create preset like
```js
module.exports = function() {
  return {
    plugins: [
      "pluginA",
      "pluginB",
      "pluginC",
    ]
  };
}
```

**Reply by nicolo-ribaudo** [reply]
Yes, that should work.

If it doesn't, convert your config file to `.js` and do this:

```js
module.exports = {
  "presets": [
    () => ({
      "plugins": ["pluginB"]
    }),
    "presetA"
  ]
}
```
